### PR TITLE
Add notebooks testing to GH Actions PR Workflow

### PR DESCRIPTION
**Pull request by ajschmidt8**
This PR is a continuation of #3076. It adds notebook testing to the PR workflow for GitHub Actions.

We'll continue to iterate as needed on this PR so that we can merge #3076 since that is holding up some other efforts.

### CODE REVIEW

**Review by Ethyling — APPROVED** [review]

**Comment by ajschmidt8** [comment]
@gpucibot merge

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -47,3 +47,13 @@ jobs:
     uses: rapidsai/shared-action-workflows/.github/workflows/conda-python-tests.yaml@main
     with:
       build_type: pull-request
+  conda-notebook-tests:
```
Comment (bdice) [inline .github/workflows/pr.yaml:50]: Should be added to pr-builder list.

File: conda/environments/all_cuda-115_arch-x86_64.yaml [hunk 2]
```diff
@@ -32,6 +32,7 @@ dependencies:
 - nccl>=2.9.9
 - networkx>=2.5.1
 - ninja
+- notebook
```
Comment (bdice) [inline conda/environments/all_cuda-115_arch-x86_64.yaml:35]: This is a duplicate pinning.

**Review by bdice — COMMENTED** [review]
@ajschmidt8 A couple fixes for a follow-up.

**Comment by ajschmidt8** [comment]
> @ajschmidt8 A couple fixes for a follow-up.

good catches. fixed in https://github.com/rapidsai/cugraph/pull/3097

**Pull request merged**